pyomo/contrib/piecewise/piecewise_linear_function.py

```diff
@@ -499,7 +573,18 @@ def _getitem_when_not_present(self, index):
                 "a list of corresponding simplices, or a dictionary "
                 "mapping points to nonlinear function values."
             )
-        return handler(self, obj, parent, nonlinear_function)
+        obj = handler(self, obj, parent, nonlinear_function)
+
+        # If the user wanted to override the triangulation tag, do it after we
+        # are finished setting it ourselves.
+        if self._triangulation_override_rule is not None:
+            triangulation_override = self._triangulation_override_rule(
+                parent, obj._index
+            )
+            if triangulation_override is not None:
+                obj._triangulation = triangulation_override
```

Comment:
Why are we supporting this? If they want to override the triangulation, can't they just specify their own simplices after doing the work themselves?